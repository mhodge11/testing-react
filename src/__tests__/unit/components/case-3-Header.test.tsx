/**
 * @jest-environment jsdom 
 */

import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import Header from '../../../components/Header'
import { FiltersContext } from '../../../context/filters'

describe('The Header component', () => {
  const defaultContextValue = {
    toggleShowingFilters: jest.fn(),
  }

  const setupHeader = (contextValue = defaultContextValue) =>
    render(
      <FiltersContext.Provider value={contextValue as any}>
        <Header />
      </FiltersContext.Provider>,
    )

  it('renders header correctly', () => {
    const { asFragment } = setupHeader()
    expect(asFragment()).toMatchSnapshot()
  })

  it('toggles the filter open when the Filter button is clicked', () => {
    const { getByText } = setupHeader()
    const filterButton = getByText('Filter')
    fireEvent.click(filterButton)
    expect(defaultContextValue.toggleShowingFilters).toHaveBeenCalled()
  })
})
