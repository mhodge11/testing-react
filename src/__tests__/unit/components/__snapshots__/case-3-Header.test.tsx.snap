// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component renders header correctly 1`] = `
<DocumentFragment>
  <header
    class="sc-hLBbgP inbbam"
  >
    <span
      class="sc-dkrFOg iRcHBj"
    />
    <div
      class="sc-bcXHqe OKlwO"
    >
      TESTING 
      <span />
      REACT.DEV
    </div>
    <button
      class="sc-gswNZR fvOAWJ"
      data-testid="FilterButton"
    >
      Filter
    </button>
  </header>
</DocumentFragment>
`;
